evalml/pipelines/component_graph.py

```diff
@@ -238,7 +238,13 @@ def _fit_transform_features_helper(self, needs_fitting, X, y=None):
             self.compute_order[:-1], X, y=y, fit=needs_fitting
         )
         final_component_inputs = []
-        for parent in self.get_parents(self.compute_order[-1]):
+
+        parent_inputs = [
+            parent_input
+            for parent_input in self.get_parents(self.compute_order[-1])
```

Comment:
We default to component name --> pass X so just checking that we're not checking a .y parent input :)